test/parallel/test-http2-too-many-streams.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+
+const Countdown = require('../common/countdown');
+const http2 = require('http2');
+const assert = require('assert');
+
+const server = http2.createServer({ settings: { maxConcurrentStreams: 1 } });
+
+let c = 0;
+
+server.on('stream', (stream) => {
+  assert.strictEqual(++c, 1);
+  stream.respond();
+  setImmediate(() => {
+    stream.end('ok');
+    assert.strictEqual(--c, 0);
+  });
+});
+
+server.listen(0, common.mustCall(() => {
+  const client = http2.connect(`http://localhost:${server.address().port}`);
+
+  const countdown = new Countdown(3, common.mustCall(() => {
+    server.close();
+    client.destroy();
+  }));
+
+  // Test that the maxConcurrentStreams setting is strictly enforced
+
+  client.on('remoteSettings', common.mustCall(() => {
+    assert.strictEqual(client.remoteSettings.maxConcurrentStreams, 1);
+
+    // This one should be ok
+    {
+      const req = client.request();
+      req.resume();
+      req.on('end', () => {
+        countdown.dec();
+
+        setImmediate(() => {
+          const req = client.request();
+          req.resume();
+          req.on('end', () => countdown.dec());
+        });
+      });
+    }
+
+    // This one should fail
+    {
+      const req = client.request();
+      req.resume();
+      // TODO(jasnell): Still investigating precisely why, but on Windows,
+      //                the error is not emitted because the underlying
+      //                mechanism ensures that only one request is sent
+      //                at a time per the maxConcurrentStreams setting.
+      //                This likely has to do with the way the response
+      //                is being handled on the server side. This is safe
+      //                to ignore on Windows because of the assert.strictEqual
+      //                check in the on('stream') handler which ensures that
+      //                only one request is being handled at any given time.
```

Comment:
I don't think this is safe to ignore. IMHO it is part of the error handling refactoring that we talked about. All client in all OS __must__ get an error in this condition. This test is about the server-side, so it's ok. If we have an issue to track the error handling refactoring, link it.

Reply:
The comment is a bit incorrect, we're not so much *ignoring* an error, we're just not *expecting* one. I'll be investigating further on why later on today, but on windows this second request is serialized out *after* the first response is received -- which is what *should* be happening. I actually think it's a bug on the *posix* side. The test is working exactly as it should. I'm not planning on landing this until I track down the specific issue.